### PR TITLE
OIDC groups and userinfo support

### PR DESCRIPTION
**Pull request by drekle**
Trying to resume 
#1315 
#1133
Also noticed #1454 is a little more relevant and okay making changes after that one if one of these can merge.

### CODE REVIEW

**Comment by drekle** [comment]
@JoelSpeed I tried to resolve many of your comments from #1315.  We would very much like to use this.  Let me know what I can do to try to get this merged.

**Comment by drekle** [comment]
@kalinon please +1 if this is okay for me to pick up.

**Comment by kalinon** [comment]
> @kalinon please +1 if this is okay for me to pick up. Otherwise I am okay with you cherry-picking anything I have done.

Please do, i appreciate it. Go is not my main language and i would rather not muck it up.

**Comment by drekle** [comment]
This is failing because I must have blown away the last PR to override the userID key.  Will reopen when resolved.

**Comment by drekle** [comment]
@srenatus any idea on the direction here with all these PR's?

**Comment by kalinon** [comment]
Looks good to me. If i could approve i would!

**Comment by JoelSpeed** [comment]
Hey, I don't have time to look now, but will try to over the weekend

File: connector/oidc/oidc.go [hunk 1]
```diff
@@ -275,10 +296,97 @@ func (c *oidcConnector) HandleCallback(s connector.Scopes, r *http.Request) (ide
...
+
+	return nil
+}
+
+// Parse the OIDC monikers and return the Common Name fields as the groups
+func (c *oidcConnector) getGroups(claims map[string]interface{}) (groups []string, err error) {
+	if rawGroup, ok := claims[c.groupsKey]; ok {
+		if groupList, ok := rawGroup.([]interface{}); ok {
+			for _, group := range groupList {
+				monikers := strings.Split(group.(string), ",")
```
Comment (drekle) [inline connector/oidc/oidc.go]: unchecked type

**Comment by nabokihms** [comment]
UserInfo endpoint was implemented in https://github.com/dexidp/dex/pull/1473 
Groups are supported by enabling [the flag](https://github.com/dexidp/dex/blob/e2ce0b6fca1c52488f32e5e3102325ec253fb623/connector/oidc/oidc.go#L44-L45). The group key is [also customizable](https://github.com/dexidp/dex/blob/e2ce0b6fca1c52488f32e5e3102325ec253fb623/connector/oidc/oidc.go#L79-L80).

Closing this PR. Please open a new one if you still have an issue.

**Pull request closed without merging**